examples/PetStore/Petstore-openApi.ps1

```diff
@@ -0,0 +1,626 @@
+                    }
+                } else {
+                    Write-PodeHtmlResponse -Value 'Pet not found' -StatusCode 404
+                }
+            } else {
+                Write-PodeJsonResponse -Value 'Invalid ID supplied' -StatusCode 400
+            }
+
+        } | Set-PodeOARouteInfo -Summary 'Find pet by ID' -Description 'Returns a single pet.' -Tags 'pet' -OperationId 'getPetById' -PassThru |
+            Set-PodeOARequest -PassThru -Parameters (
+                New-PodeOAIntProperty -Name 'petId' -Description 'ID of pet to return'  -Format Int64 |
+                    ConvertTo-PodeOAParameter -In Path -Required ) |
+                Add-PodeOAResponse -StatusCode 200 -Description 'Successful operation' -Content  (New-PodeOAContentMediaType -ContentMediaType 'application/json', 'application/xml' -Content 'Pet') -PassThru |
+                Add-PodeOAResponse -StatusCode 400 -Description 'Invalid ID supplied' -PassThru |
+                Add-PodeOAResponse -StatusCode 404 -Description 'Pet not found' -PassThru |
+                Add-PodeOAResponse -StatusCode 415
+
+
+        <#
+            POST '/pet/{petId}'
+        #>
+
+        Add-PodeRoute -PassThru -Method post -Path '/pet/:petId' -Authentication 'petstore_auth' -Scope 'write:pets', 'read:pets' -ScriptBlock {
+            $petId = $WebEvent.Parameters['petId']
+            $name = $WebEvent.Query['name']
+            $status = $WebEvent.Query['status']
+
+            if ($petId -and (Test-Pet -Id $petId)) {
+                if (Update-Pet -Id $petId -Name $name -Status $status) {
+                    Save-PodeState -Path $using:PetDataJson
+                } else {
+                    Write-PodeHtmlResponse -StatusCode 405 -Value 'Invalid Input'
+                }
+            } else {
+                Write-PodeHtmlResponse -StatusCode 405 -Value 'Invalid Input'
+            }
+        } | Set-PodeOARouteInfo -Summary 'Updates pet with ID' -Description 'Updates a pet in the store with form data' -Tags 'pet' -OperationId 'updatePetWithForm' -PassThru |
+            Set-PodeOARequest -PassThru -Parameters  ( New-PodeOAIntProperty -Name 'petId' -Description 'ID of pet that needs to be updated'  -Format Int64 |
+                    ConvertTo-PodeOAParameter -In Path -Required ),
+                                    (  New-PodeOAStringProperty -Name 'name' -Description 'Name of pet that needs to be updated' | ConvertTo-PodeOAParameter -In Query ) ,
+                                    (  New-PodeOAStringProperty -Name 'status' -Description 'Status of pet that needs to be updated' | ConvertTo-PodeOAParameter -In Query ) |
+                Add-PodeOAResponse -StatusCode 405 -Description 'Invalid Input'
+
+        <#
+            DELETE '/pet/{petId}'
+        #>
+        Add-PodeRoute -PassThru -Method Delete -Path '/pet/:petId' -Authentication 'merged_auth_All' -Scope 'write:pets', 'read:pets' -ScriptBlock {
```

Comment:
I have an issue with these merged authentications. How does it work when you have multiple auth, and all of them must be satisfied?

Reply:
When a merged auth method is set so all auth methods specified must be satisfied, say API key and Basic auth methods, then each method is called in turn - in the order passed - and must pass their own validation. If the first, API key in this case, fails then the whole merged auth method fails - Basic auth is not attempted.

The same merged auth, but with only need one to be satisfied, is done in the same manner but if 1 fails the next method is called. If at least 1 passes then the merged auth is successful, but if all fail then the merged auth fails.

Reply:
please try `Broken-petstore_Auth-Petstore-openApi.ps1` and you will see the issue with -OAuth2 when the configuration point to something that does not exist or is offline